File: 31-60/groupAnagram49.py
from collections import defaultdict
from typing import List

def groupAnagrams(strs) -> list:
    # this is a dictionary for key and value to store to it
    # It's gonna be look like this: {(1, 0, ..., 1): ["eat", "tea", "ate"]}
    res = defaultdict(list)
    for s in strs:
        count = [0] * 26
        for c in s:
            count[ord(c) - ord('a')] += 1
        res[tuple(count)].append(s)

    return res.values()


strs=["eat","tea","tan","ate","nat","bat"]

groupAnagrams(strs)

File: 31-60/groupAnagram49.js

function groupAnagram49(str) {
    let res = {}
    for (let s of str) {
        let zeros = Array(26).fill(0);
        for (let c of s) zeros[c.charCodeAt(0) - 'a'.charCodeAt(0)] +=1;

        let key = zeros.join(',')
        if (!res[key]) res[key] = []
        res[key].push(s);
    }
    return Object.values(res);
}

let strs=["eat","tea","tan","ate","nat","bat"]

console.log(groupAnagram49(strs))